src/main.ts

```diff
@@ -118,6 +120,20 @@ for await (const app of apps) {
       clientSecret: app.clientSecret,
     });
 
+    if (app.appRoles) {
+      await addAppRoles({
+        token: token,
+        applicationId: applicationId,
+        appRoles: app.appRoles,
+      });
+
+      // Pass the object ID of the enterprise app
+      await addAppRoleGroupAssignmentsToApp({
+        token: token,
+        applicationId: servicePrincipalObjectId,
```

Comment:
as the `servicePrincipalObjectId` is passed in, does it mean we will need to update the `servicePrincipal` object to add/update app roles? as per my above comment.

https://learn.microsoft.com/en-us/graph/api/serviceprincipal-update?view=graph-rest-1.0&tabs=http#http-request

Reply:
That's what currently happens here yes - the groups are assigned x roles on the service principal/enterprise app